mottak/src/main/kotlin/no/nav/melosys/soknadmottak/soknad/SoknadSkjemaOversetter.kt

```diff
@@ -0,0 +1,240 @@
+        if (skipListe == null) return null
+        return SkipListe(skipListe.skip.map { skip ->
+            Skip(
+                skip.fartsomraade.value(),
+                skip.skipNavn,
+                skip.flaggland,
+                skip.territorialEllerHavnLand
+            )
+        })
+    }
+
+    private fun oversettArbeidstaker(innhold: Innhold): Arbeidstaker {
+        val arbeidstaker = innhold.arbeidstaker
+        return Arbeidstaker(
+            oversettMedfølgendeBarn(innhold),
+            arbeidstaker.isReiserMedBarnTilUtlandet,
+            arbeidstaker.fulltNavn,
+            arbeidstaker.foedselsnummer,
+            arbeidstaker.foedeland,
+            arbeidstaker.foedested,
+            arbeidstaker.utenlandskIDnummer
+        )
+    }
+
+    private fun oversettMedfølgendeBarn(innhold: Innhold): List<BarnMed> {
+        val barn = innhold.arbeidstaker.barn
+        if (barn == null || barn.barnet == null) return emptyList()
+        return barn.barnet.map { b -> BarnMed(b.foedselsnummer, b.navn) }
+    }
+
+    private fun oversettKontaktperson(innhold: Innhold): Kontakperson? {
+        val kontaktperson = innhold.arbeidsgiver.kontaktperson ?: return null
+        return Kontakperson(
+            kontaktperson.kontaktpersonNavn,
+            kontaktperson.kontaktpersonTelefon,
+            oversettAnsattHos(innhold),
+            arbeidstakerHarGittFullmakt(innhold),
+            hentKontaktVirksomhetsnummer(innhold),
+            hentKontaktVirksomhetsnavn(innhold)
+        )
+    }
+
+    private fun oversettAnsattHos(innhold: Innhold): String {
+        return if (erRådgivningsfirmaFullmektig(innhold)) {
+            "I et rådgivningsfirma"
+        } else {
+            "Hos arbeidsgiveren"
+        }
+    }
+
+    private fun oversettUtenlandsoppdrag(innhold: Innhold): Utenlandsoppdrag {
+        val midlertidigUtsendt = innhold.midlertidigUtsendt
+        val utenlandsoppdraget = midlertidigUtsendt.utenlandsoppdraget
+        return Utenlandsoppdrag(
+            innhold.midlertidigUtsendt.arbeidsland,
+            oversettTidsrom(utenlandsoppdraget.periodeUtland),
+            utenlandsoppdraget.isErstatterTidligereUtsendte,
+            oversettTidsrom(utenlandsoppdraget.samletUtsendingsperiode),
+            utenlandsoppdraget.isSendesUtOppdragIUtlandet,
+            utenlandsoppdraget.isDrattPaaEgetInitiativ,
+            utenlandsoppdraget.isAnsattForOppdragIUtlandet,
+            utenlandsoppdraget.isAnsattEtterOppdraget
+        )
+    }
+
+    private fun oversettTidsrom(tidsrom: Tidsrom?): String {
+        if (tidsrom == null) return ""
+        val fom = oversettDato(tidsrom.periodeFra)
+        val tom = oversettDato(tidsrom.periodeTil)
+        return "f.o.m. $fom t.o.m. $tom"
+    }
+
+    private fun oversettDato(calendar: XMLGregorianCalendar?): String {
+        if (calendar == null) return ""
+        return "${calendar.year}-${calendar.month}-${calendar.day}"
+    }
```

Comment:
```kotlin
calendar?.let {
    "${calendar.year}-${calendar.month}-${calendar.day}"
} ?: ""
```
er kanskje mer Kotlin-ish (gjelder også flere steder). Ser at denne ikke nødvendigvis gjør det mer lesbart, så opp til deg om du føler det er verdt bryet.

Reply:
> Bare litt pirk angående kodestil i Kotlin, ellers ser det veldig bra ut 👍

Som begynner i Kotlin er jeg veldig takknemlig for slike tilbakemeldinger.

Reply:
Helt enig i at det er mer kotlinsk. Har spørsmål dog. Bør det ikke være:
```
calendar?.let {
    "${it.year}-${it.month}-${it.day}"
} ?: ""
``` 
?
Hvordan får du farger i koden din?

Reply:
Ja, det har du rett i 😄 

> Hvordan får du farger i koden din?

Starter kodeblokken med ```kotlin